installscripts/jazz-terraform-unix-noinstances/chefProvisioner.tf

```diff
@@ -128,3 +166,16 @@ resource "null_resource" "postJenkinsConfiguration" {
     command = "${var.injectingBootstrapToJenkinsfiles_cmd} ${lookup(var.scmmap, "scm_elb")} ${lookup(var.scmmap, "scm_type")}"
   }
 }
+
+resource "null_resource" "postJenkinsConfiguration_gitlab" {
+  count = "${var.scmgitlab}"
+  depends_on = ["null_resource.configureJenkinsInstance", "null_resource.configureJenkinsDocker", "aws_elasticsearch_domain.elasticsearch_domain", "null_resource.configureCliJenkins_dockerized"]
```

Comment:
Updated. The reason why it has been split out is, 

For senario3, we have gitlab and jenkins URLs, created on run time unlike for scenario1. First I thought terraform would support ternary conditions like,

  provisioner "local-exec" {
    command = "${var.injectingBootstrapToJenkinsfiles_cmd} ${var.dockerizedJenkins  == 1 ? aws_lb.alb_ecs_gitlab.dns_name : ${lookup(var.jenkinsservermap, "jenkins_elb") } ${lookup(var.scmmap, "scm_type")}"
  }

But it has been failed, since for scenario1, it will throw Resource not found exception for "aws_lb.alb_ecs_gitlab.dns_name".

Reply:
Updated.